pandas/core/missing.py

```diff
@@ -7,65 +7,53 @@
 import numpy as np
 
 from pandas._libs import algos, lib
-from pandas._typing import DtypeObj
+from pandas._typing import ArrayLike, DtypeObj
 from pandas.compat._optional import import_optional_dependency
 
 from pandas.core.dtypes.cast import infer_dtype_from_array
 from pandas.core.dtypes.common import (
     ensure_float64,
     is_integer_dtype,
     is_numeric_v_string_like,
-    is_scalar,
     needs_i8_conversion,
 )
 from pandas.core.dtypes.missing import isna
 
 
-def mask_missing(arr, values_to_mask):
+def mask_missing(arr: ArrayLike, values_to_mask) -> np.ndarray:
     """
     Return a masking array of same size/shape as arr
     with entries equaling any member of values_to_mask set to True
-    """
-    dtype, values_to_mask = infer_dtype_from_array(values_to_mask)
 
-    try:
-        values_to_mask = np.array(values_to_mask, dtype=dtype)
+    Parameters
+    ----------
+    arr : ArrayLike
+    values_to_mask: list, tuple, or scalar
 
-    except Exception:
-        values_to_mask = np.array(values_to_mask, dtype=object)
+    Returns
+    -------
+    np.ndarray[bool]
+    """
+    # When called from Block.replace/replace_list, values_to_mask is a scalar
+    #  known to be holdable by arr.
+    # When called from Series._single_replace, values_to_mask is tuple or list
+    dtype, values_to_mask = infer_dtype_from_array(values_to_mask)
```

Comment:
ok, this is very confusing and ought to simplify and tighten up the signature here (to make it consistently called), but clearly for a followon.